test/helpers/name.go

```diff
@@ -75,9 +79,11 @@ func AppendRandomString(prefix string) string {
 // RandomString will generate a random string.
 func RandomString() string {
 	suffix := make([]byte, randSuffixLen)
+	nameMu.Lock()
 	for i := range suffix {
 		suffix[i] = letterBytes[nameRand.Intn(len(letterBytes))]
 	}
+	nameMu.Unlock()
```

Comment:
minor tweak - let's just do a defer unlock

```suggestion
	nameMu.Lock()
	defer nameMu.Unlock()
	for i := range suffix {
		suffix[i] = letterBytes[nameRand.Intn(len(letterBytes))]
	}

```